core/runtime/src/main/java/io/quarkus/runtime/configuration/PropertiesUtil.java

```diff
@@ -121,23 +49,27 @@ public static boolean isPropertyQuarkusCompoundName(NameIterator propertyName) {
         return propertyName.getName().startsWith("\"quarkus.");
     }
 
-    /**
-     * Removes false positives of configuration properties marked as unknown. To populate the old @ConfigRoot, all
-     * properties are iterated and matched against known roots. With @ConfigMapping the process is different, so
-     * properties that are known to @ConfigMapping are not known to the @ConfigRoot, so they will be marked as being
-     * unknown. It is a bit easier to just double-check on the unknown properties and remove these false positives by
-     * matching them against the known properties of @ConfigMapping.
-     *
-     * @param unknownProperties the collected unknown properties from the old @ConfigRoot mapping
-     * @param filterPatterns the mapped patterns from the discovered @ConfigMapping
-     */
-    public static void filterUnknown(Set<String> unknownProperties, KeyMap<Boolean> filterPatterns) {
-        Set<String> toRemove = new HashSet<>();
-        for (String unknownProperty : unknownProperties) {
-            if (filterPatterns.hasRootValue(unknownProperty)) {
-                toRemove.add(unknownProperty);
+    public static Set<PropertyName> toPropertyNames(final Set<String> names) {
+        Map<PropertyName, String> propertyNames = new HashMap<>();
+        for (String name : names) {
+            PropertyName propertyName = new PropertyName(name);
```

Comment:
What it...`PropertyName` become a mutable class (which can wrap different names) and we allocate a single instance of it?

Reply:
We only need this at start to validate names. Adding is only done once. My plan was to write a proper API for this in SR Config.